Error: Unresolved import: `clap::Clap`

Hi, thank you for sharing this repo. Unfortunately, I'm not able to run the code.

## Version error
When I try to execute: `cargo run -- --help`, I get the following errors:
```rust
error[E0432]: unresolved import `clap::Clap`
 --> kradical_converter/src/main.rs:1:12
  |
1 | use clap::{Clap, Parser};
  |            ^^^^ no `Clap` in the root

... (more clap errors)
```

I noticed that the dependency in `kradical_converter/Cargo.toml` is set to:
```toml
[dependencies]
clap = "3.0.0-beta.2"
```
but it actually installs version `3.2.25`. By adding a `=`, it installs the version specified and I get a step further:
```toml
[dependencies]
clap = "=3.0.0-beta.2"
```

## Error: Unresolved reference: `Clap` in `opts.rs`
Now in `kradical_converter/src/opts`, the following error appears:
```rust
error: cannot find derive macro `Clap` in this scope
 --> kradical_converter/src/opts.rs:3:10
  |
3 | #[derive(Clap, Clone, PartialEq, Eq, Debug)]
  |          ^^^^
  |
note: `Clap` is imported here, but it is only a trait, without a derive macro
 --> kradical_converter/src/opts.rs:1:25
  |
1 | use clap::{AppSettings, Clap};
  |                         ^^^^
```

I added `"derive"` as by this StackOverflow issue: https://stackoverflow.com/questions/69863208/clap-options-with-derive-macros-fails-to-compile-in-rust-clap-3-0-0-beta-5
```toml
[dependencies]
clap = { version = "=3.0.0-beta.2", features = ["derive"] }
```
but it doesn't resolve the issue.

Any idea where it goes wrong?

--- a/README.md
+++ b/README.md
@@ -3,6 +3,55 @@
 Utilities for working with the [Electronic Dictionary Research and Development Group](https://www.edrdg.org/) (EDRDG) [radical decomposition](https://www.edrdg.org/krad/kradinf.html) files.
 
 
+## Usage
+
+### Converting EDRDG Files
+
+The `kradical_converter` tool can convert EDRDG radical files to various formats. Use multiple `--inputs` flags to process multiple files together.
+
+The JSON and UTF8 txt files are created under `/assets/outputs/` with the below commands.
+
+#### Create JSON
+With output extension `.json` and `json` option:
+```bash
+# Generate krad.json
+cargo run -p kradical_converter -- --inputs ./assets/edrdg_files/kradfile --inputs ./assets/edrdg_files/kradfile2 --output ./assets/outputs/krad.json krad json
+
+# Generate radk.json
+cargo run -p kradical_converter -- --inputs ./assets/edrdg_files/radkfile --inputs ./assets/edrdg_files/radkfile2 --output ./assets/outputs/radk.json radk json
+```
+
+#### Create UTF-8 txt
+With output extension `.txt` and `unicode` option:
+```bash
+# Generate krad_utf8.txt (kanji → radicals mapping)
+cargo run -p kradical_converter -- --inputs ./assets/edrdg_files/kradfile --inputs ./assets/edrdg_files/kradfile2 --output ./assets/outputs/krad_utf8.txt krad unicode
+
+# Generate radk_utf8.txt (radical → kanji mapping)
+cargo run -p kradical_converter -- --inputs ./assets/edrdg_files/radkfile --inputs ./assets/edrdg_files/radkfile2 --output ./assets/outputs/radk_utf8.txt radk unicode
+```
+
+#### Create Rust static files for `kradical_static`
+With output extension `.rs` and `rust` option:
+```bash
+# Generate decompositions.rs (for kradical_static crate)
+cargo run -p kradical_converter -- --inputs ./assets/edrdg_files/kradfile --inputs ./assets/edrdg_files/kradfile2 --output ./kradical_static/src/decompositions.rs krad rust
+
+# Generate memberships.rs (for kradical_static crate)
+cargo run -p kradical_converter -- --inputs ./assets/edrdg_files/radkfile --inputs ./assets/edrdg_files/radkfile2 --output ./kradical_static/src/memberships.rs radk rust
+```
+
+#### Executable usage
+1. First, build the release version of the converter:
+
+```bash
+cargo build --release
+```
+
+2. Then replace `cargo run -p kradical_converter -- --inputs` with `./target/release/kradical_converter --inputs` in the above commands.
+    - On Windows also replace `/` with `\` for the path.
+
+
 ## Crates
 
 More information about each crate included with the project in the associated readme files:
@@ -32,4 +81,4 @@ Included with this project under `assets/outputs` are several UTF-8-encoded vari
 
 In accordance with the [EDRDG license statement](http://www.edrdg.org/edrdg/licence.html), this project is distributed under the [Attribution-ShareAlike 3.0 Unported](https://creativecommons.org/licenses/by-sa/3.0/legalcode) license. The files included under `assets/edrdg_files` were downloaded from the [Monash Nihongo FTP Archive](http://ftp.edrdg.org/pub/Nihongo/00INDEX.html#dic_fil) and are the property of EDRDG.
 
-JIS X 0212 conversion tables are distributed under the [Unicode license](http://www.unicode.org/copyright.html). 
+JIS X 0212 conversion tables are distributed under the [Unicode license](http://www.unicode.org/copyright.html).

--- a/kradical_converter/Cargo.toml
+++ b/kradical_converter/Cargo.toml
@@ -9,6 +9,6 @@ keywords = ["japanese", "kanji", "radical"]
 categories = ["command-line-utilities", "encoding", "parsing"]
 
 [dependencies]
-clap = "3.0.0-beta.2"
+clap = { version = "4.0", features = ["derive"] }
 thiserror = "1"
 kradical_parsing = "0.1.0"

--- a/kradical_converter/README.md
+++ b/kradical_converter/README.md
@@ -3,9 +3,10 @@
 [![LICENSE](https://img.shields.io/crates/l/kradical_converter)](https://crates.io/crates/kradical_converter)
 [![Crates.io Version](https://img.shields.io/crates/v/kradical_converter)](https://crates.io/crates/kradical_converter)
 
-A binary for converting the original JIS-encoded files to other formats. It is also able to combine multiple of these files. Below is an example invocation. For more information, use `cargo run -- --help`. 
+A binary for converting the original JIS-encoded files to other formats. It is also able to combine multiple of these files. Below is an example invocation. For more information, use `cargo run -- --help`.
 
-`kradical_converter radk unicode --inputs .\assets\edrdg_files\radkfile .\assets\edrdg_files\radkfile2 --output .\assets\outputs\radk_utf8.txt`
+Below command assumes Windows. Use `/` on Linux / WSL:
+`kradical_converter radk unicode --inputs .\assets\edrdg_files\radkfile --inputs .\assets\edrdg_files\radkfile2 --output .\assets\outputs\radk_utf8.txt`
 
 
 ## License

--- a/kradical_converter/src/main.rs
+++ b/kradical_converter/src/main.rs
@@ -1,4 +1,4 @@
-use clap::Clap;
+use clap::Parser;
 use error::ConvertError;
 use std::{fs::OpenOptions, io::Write};
 

--- a/kradical_converter/src/opts.rs
+++ b/kradical_converter/src/opts.rs
@@ -1,28 +1,30 @@
-use clap::{AppSettings, Clap};
+use clap::{Parser, ValueEnum};
 
-#[derive(Clap, Clone, PartialEq, Eq, Debug)]
-#[clap(setting = AppSettings::ColoredHelp)]
+#[derive(Parser, Clone, PartialEq, Eq, Debug)]
+#[command(about = "Convert EDRDG radical decomposition files to various formats")]
 pub struct Opts {
-    #[clap(arg_enum)]
+    /// Input format (krad or radk)
     pub input_format: InputFormat,
 
-    #[clap(arg_enum)]
+    /// Output format (unicode, rust, or json)
     pub output_format: OutputFormat,
 
-    #[clap(short, long, required = true)]
+    /// Input files to process
+    #[arg(short, long, required = true)]
     pub inputs: Vec<String>,
 
-    #[clap(short, long)]
+    /// Output file path
+    #[arg(short, long)]
     pub output: String,
 }
 
-#[derive(PartialEq, Eq, Clone, Copy, Clap, Debug)]
+#[derive(PartialEq, Eq, Clone, Copy, ValueEnum, Debug)]
 pub enum InputFormat {
     Radk,
     Krad,
 }
 
-#[derive(PartialEq, Eq, Clone, Copy, Clap, Debug)]
+#[derive(PartialEq, Eq, Clone, Copy, ValueEnum, Debug)]
 pub enum OutputFormat {
     Unicode,
     Rust,

